packages/origin-247-certificate/src/offchain-certificate/events/Certificate.events.ts

```diff
@@ -31,78 +53,190 @@ export class CertificateIssuedEvent<MetadataType = unknown> implements ICertific
     public readonly version = version;
     public readonly createdAt = new Date();
 
-    constructor(
+    private constructor(
+        private readonly _id: number,
         public readonly internalCertificateId: number,
         public readonly payload: IIssueCommand<MetadataType>
     ) {}
+
+    public get id() {
```

Comment:
So, generally ICertificateEvent should store its id,
but it's not available before we save this event.

So that's the reason for this code.